src/compiler/bundle/ext-transforms-plugin.ts

```diff
@@ -7,6 +7,17 @@ import { parseImportPath } from '../transformers/stencil-import-path';
 import type { Plugin } from 'rollup';
 import { runPluginTransformsEsmImports } from '../plugin/plugin';
 
+/**
+ * A Rollup plugin which bundles up some transformation of CSS imports as well
+ * as writing CSS files for components to disk for the `DIST_COLLECTION` output
+ * target.
```

Comment:
tbh I'm not sure, I thought that the only file writing going on here was with the return values of `runPluginTransformsEsmImports` which I _think_ only has to do with CSS files? But unfortunately as always most of this stuff is not really labeled so clearly :/

I'll change the comment to leave it a bit more ambiguous for now, and maybe something in the future we can do a more thorough bit of archeology on this region of the codebase

Reply:
FWIW - added a little `console.log` here to confirm:
```diff
diff --git a/src/compiler/bundle/ext-transforms-plugin.ts b/src/compiler/bundle/ext-transforms-plugin.ts
index 08e955ef5..cdf37009b 100644
--- a/src/compiler/bundle/ext-transforms-plugin.ts
+++ b/src/compiler/bundle/ext-transforms-plugin.ts
@@ -39,6 +39,7 @@ export const extTransformsPlugin = (
      * @returns metadata for Rollup or null if no transformation should be done
      */
     async transform(_, id) {
+      console.log(`id: ${id}`);
       if (/\0/.test(id)) {
         return null;
       }
```

Then built the Stencil, with the installed it in the output of the Stencil CLI and ran `npm run build`:

```
id: my-component.entry
id: core
id: MyComponent
id: @lazy-browser-entrypoint?app-data=conditional
id: @lazy-external-entrypoint?app-data=conditional
id: /path/to/project/intellij-test/src/index.ts
id: @stencil/core/internal/app-globals
id: @stencil/core/internal/app-globals
id: /path/to/project/intellij-test/src/index.ts
id: /path/to/stencil/internal/client/patch-browser.js
id: @stencil/core/internal/app-data
id: /path/to/stencil/internal/client/patch-esm.js
id: /path/to/stencil/internal/client/index.js?app-data=conditional
id: /path/to/stencil/internal/client/css-shim.js
id: /path/to/stencil/internal/client/dom.js
id: /path/to/stencil/internal/client/shadow-css.js
id: /path/to/project/intellij-test/src/components/my-component/my-component.tsx
id: /path/to/project/intellij-test/src/components/my-component/my-component.tsx
id: /path/to/project/intellij-test/src/utils/utils.ts
id: /path/to/project/intellij-test/src/utils/utils.ts
id: /path/to/project/intellij-test/src/components/my-component/my-component.css?tag=my-component&encapsulation=shadow
id: /path/to/project/intellij-test/src/components/my-component/my-component.css?tag=my-component&encapsulation=shadow
```

Reply:
I believe a rollup plugin like this will end up being called with all the modules in the build, my reasoning was that in some of the logic down below it doesn't appear as if it would write those files except for CSS files, but I could be wrong about that

Reply:
Bah - sorry, that's my bad 🤦 Totally misunderstood!

Reply:
this bit is what I'm referring to:

https://github.com/ionic-team/stencil/blob/103ec6098a5367d0de26450e1010bddb7ae8e890/src/compiler/bundle/ext-transforms-plugin.ts#L36-L52

If I make a little change like this:

```diff
diff --git a/src/compiler/bundle/ext-transforms-plugin.ts b/src/compiler/bundle/ext-transforms-plugin.ts
index 08e955ef5..fdf9c9d0a 100644
--- a/src/compiler/bundle/ext-transforms-plugin.ts
+++ b/src/compiler/bundle/ext-transforms-plugin.ts
@@ -91,6 +91,7 @@ export const extTransformsPlugin = (
             await Promise.all(
               collectionDirs.map(async (outputTarget) => {
                 const collectionPath = join(outputTarget.collectionDir, relPath);
+                console.log(`extTransformsPlugin::writing to ${collectionPath}`);
                 await compilerCtx.fs.writeFile(collectionPath, pluginTransforms.code);
               })
             );
```

and run a build in Framework I only see CSS files there:

<img width="1360" alt="Screen Shot 2022-08-15 at 12 43 59 PM" src="https://user-images.githubusercontent.com/6207644/184706056-14a14695-7734-4939-b40a-31196fc56748.png">

Reply:
(more omitted for the sake of brevity!)

Reply:
@rwaskiewicz no worries!